superset/models/core.py

```diff
@@ -761,24 +761,26 @@ def wrapper(*args, **kwargs):
             post_data = request.form or {}
             d.update(post_data)
             d.update(kwargs)
-            slice_id = d.get('slice_id', 0)
+            slice_id = d.get('slice_id') 
+            form_data = d.get('form_data')
+
             try:
-                slice_id = int(slice_id) if slice_id else 0
-            except ValueError:
+                slice_id = int(d.get('slice_id') or json.loads(d.get('form_data')).get('slice_id'))
```

Comment:
Would yo prefer that I break it out in the if else style I had before that doesn't use exception for flow control? I am trying out the ValueError and TypeError approach now.